components/AddDonation.js

```diff
@@ -0,0 +1,136 @@
+import styled from "styled-components";
+import { useState } from "react";
+import TextLengthCounter from "./Counter";
+import categoryData from "../helper/category.json";
+
+export default function AddDonation({ createDonation }) {
+  const [isFormVisible, setIsFormVisible] = useState(false);
+  const [count, setCount] = useState(0);
+
+  function handleToggleForm() {
+    setIsFormVisible(!isFormVisible);
+  }
+  function handleSubmit(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const newDonationData = Object.fromEntries(formData);
+
+    createDonation(newDonationData);
+    setCount(0);
+    event.target.reset();
+    setIsFormVisible(!isFormVisible);
+  }
+
+  return (
+    <StyledDiv>
+      {!isFormVisible ? (
+        <StyledButton onClick={handleToggleForm}>
+          Neue Spende hinzufügen
+        </StyledButton>
+      ) : (
+        <StyledForm onSubmit={handleSubmit}>
+          <label htmlFor="category">Kategorie:</label>
+          <select name="category" id="category" required>
+            <option value="">-Wähle bitte eine Kategorie-</option>
+            {categoryData.map((category) => (
+              <option key={category.id} value={category.category}>
+                {category.category}
+              </option>
+            ))}
+          </select>
+
+          <label htmlFor="description">Beschreibung:</label>
+          <textarea
+            type="text"
+            id="description"
+            name="description"
+            cols="30"
+            rows="5"
+            maxLength="250"
+            pattern="[a-zA-ZäüöÄÜÖß0-9 _-+&*.,:;!?()%€]"
+            onChange={(event) => setCount(event.target.value.length)}
+            //required
+          />
+          <TextLengthCounter max={250} counter={count} />
+
+          <label htmlFor="bestBefore">Haltbar bis:</label>
+          <input type="date" id="bestBefore" name="bestBefore" />
+
+          <label htmlFor="zipCode">Postleitzahl:</label>
+          <input
+            type="text"
+            id="zipCode"
+            name="zipCode"
+            pattern="^[0-9][0-9][0-9][0-9][0-9]$"
```

Comment:
Hey Mina, instead of type=text you could also use type=number, there you dont need the pattern and could just set a min={1} and max={99999} number. Its useful for inputs where you only want to use numbers :)

Reply:
changed it - min{11111} and max={99999} since it has to be 5 digits.